go/libraries/doltcore/servercfg/yaml_config.go

```diff
@@ -249,7 +247,113 @@ func clusterConfigAsYAMLConfig(config ClusterConfig) *ClusterYAMLConfig {
 
 // String returns the YAML representation of the config
 func (cfg YAMLConfig) String() string {
-	data, err := yaml.Marshal(cfg)
+	return formattedYAMLMarshal(cfg)
+}
+
+// Behaves like String, but includes empty fields instead of omitting them.
+// If an empty field has a default value, the default will be used.
+// If an empty field has no default value, a commented-out placeholder will be used.
+func (cfg YAMLConfig) VerboseString() string {
+	withDefaults := cfg
+	withDefaults.fillDefaults()
+
+	formatted := formattedYAMLMarshal(removeOmitemptyTags(withDefaults))
+	formatted = commentNullYAMLValues(formatted)
+
+	return formatted
+}
+
+// Assumes YAMLConfig has no circular references.
+func (cfg *YAMLConfig) fillDefaults() {
+	defaults := defaultServerConfigYAML()
+	recursiveFillDefaults(reflect.ValueOf(cfg), reflect.ValueOf(defaults))
+}
+
+func recursiveFillDefaults(cfgValue, defaultsValue reflect.Value) {
```

Comment:
Like James said, I think we would prefer to just explicitly list all the fields here, rather than use reflection.